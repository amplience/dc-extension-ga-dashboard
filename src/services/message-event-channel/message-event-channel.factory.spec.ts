import type { ConnectionOptions } from 'message-event-channel';
import createConnection from './message-event-channel.factory';

jest.mock('message-event-channel', () => {
  return {
    ClientConnection: function (options?: ConnectionOptions) {
      expect(options).toMatchSnapshot();
    },
  };
});

describe('message event channel factory', () => {
  it('should return a new client connection with default params', () => {
    expect(createConnection()).toMatchInlineSnapshot(`ClientConnection {}`);
  });

  it('should return a new client connection with default params', () => {
    const options = {
      targetOrigin: 'target',
      onload: true,
      timeout: 1000,
      debug: true,
      connectionTimeout: 1000,
      clientInitiates: false,
    };
    expect(createConnection(options)).toMatchInlineSnapshot(
      `ClientConnection {}`
    );
  });
});
