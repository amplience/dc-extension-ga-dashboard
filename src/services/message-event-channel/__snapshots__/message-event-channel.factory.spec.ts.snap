// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message event channel factory should return a new client connection with default params 1`] = `undefined`;

exports[`message event channel factory should return a new client connection with default params 3`] = `
Object {
  "clientInitiates": false,
  "connectionTimeout": 1000,
  "debug": true,
  "onload": true,
  "targetOrigin": "target",
  "timeout": 1000,
}
`;
